client/state/themes/actions.js

```diff
@@ -609,7 +611,8 @@ function transferStatusFailure( siteId, transferId, error ) {
 export function pollThemeTransferStatus( siteId, transferId, interval = 3000, timeout = 180000 ) {
 	const endTime = Date.now() + timeout;
 	return dispatch => {
-		const pollStatus = ( resolve, reject ) => {
+		dispatch( requestAutomatedTransferStatus( siteId ) );
+		const pollStatus = resolve => {
 			if ( Date.now() > endTime ) {
 				// timed-out, stop polling
 				dispatch( transferStatusFailure( siteId, transferId, 'client timeout' ) );
```

Comment:
I haven't yet taken care of the error and closing states. It's worth noting that the successful response from the status API includes a timestamp. If we watched on those responses and compared the given timestamp to the current time then we would have a timeout that persists across browser sessions. I plan on exploring this route.